### PR TITLE
mon/MDSMonitor: enforce mds_join_fs cluster affinity

### PR DESCRIPTION
**Pull request by batrick**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by batrick** [comment]
@liewegas this required quite a bit of rework of the bookkeeping. Notably, `standby_daemon_fscid` is removed which is why I asked to delay deployment of Octopus to the LRC.

I'll run this through some testing tonight.

**Comment by batrick** [comment]
Tagging @ceph/dashboard and @tchaikov on the bits I changed to the qa suite in `qa/tasks/mgr` and `qa/vstart_runner.py`.

**Comment by batrick** [comment]
Going to add some docs in a follow-up commit.

File: qa/tasks/vstart_runner.py [hunk 1]
```diff
@@ -1016,12 +1016,6 @@ def __init__(self, ctx):
     def admin_remote(self):
         return LocalRemote()
 
-    def set_config_opt(self, section, opt, val):
```
Comment (tchaikov) [inline qa/tasks/vstart_runner.py]: ack. so we are moving this functionality from `CephCluster` side into `CephTestCase`.

**Comment by batrick** [comment]
jenkins render docs

**Comment by batrick** [comment]
http://pulpito.ceph.com/?branch=wip-pdonnell-testing-20200211.032856

**Comment by batrick** [comment]
I'm debugging some of the failures now...

**Comment by batrick** [comment]
http://pulpito.ceph.com/?branch=wip-pdonnell-testing-20200211.222321

**Comment by batrick** [comment]
http://pulpito.ceph.com/pdonnell-2020-02-13_18:27:35-fs-wip-pdonnell-testing-20200213.155332-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
https://github.com/ceph/ceph/pull/33298#issuecomment-586258755 related?

**Comment by tchaikov** [comment]
should be addressed by #33325

**Pull request merged**